src/librustc_mir/borrow_check/error_reporting.rs

```diff
@@ -298,9 +300,9 @@ impl<'cx, 'gcx, 'tcx> MirBorrowckCtxt<'cx, 'gcx, 'tcx> {
             "immutable",
             "mutable",
         ) {
-            (BorrowKind::Shared, lft, _, BorrowKind::Mut { .. }, _, rgt)
-            | (BorrowKind::Mut { .. }, _, lft, BorrowKind::Shared, rgt, _) => tcx
```

Comment:
hmm. If my memory is right, the reason this was structured as a match on a tuple was to allow collapsing multiple cases like these and selecting the right string to use by setting `lft` and `rgt` to the appropriate string literal from the match input.

Reading over the code now, I am thinking that there is potentially no longer *any* instance of two arms being collapsed, which means we really should consider cleaning up the code by getting rid of those string literals in the tuple (and instead just pass the appropriate strings below), I think...

That cleanup doesn't have to be part of this PR, though.